docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc

```diff
@@ -0,0 +1,230 @@
+  }
+}
+
+POST /index_1/_close
+
+POST /index_2/_close
+
+POST /index_3/_close
+
+POST /index_4/_close
+
+----
+// TESTSETUP
+////
+
+[source,console]
+----
+POST /_snapshot/my_repository/my_snapshot/_restore
+----
+// TEST[s/_restore/_restore?wait_for_completion=true/]
+
+[[restore-snapshot-api-request]]
+==== {api-request-title}
+
+`POST /_snapshot/<repository>/<snapshot>`
+
+[[restore-snapshot-api-desc]]
+==== {api-description-title}
+
+Use the restore snapshot API to restore a snapshot of a cluster, including all data streams and indices in the snapshot. If you do not want to restore the entire snapshot, you can select specific data streams or indices to restore.
+
+NOTE: You cannot restore a data stream if a stream with the same name already
+exists.
+
+You can run the restore operation only on a functioning cluster. Existing indices can only be restored if they are <<indices-close,closed>> and have the same number of shards as the indices in the snapshot. The restore operation automatically opens restored indices if they were closed and creates new indices if they do not exist in the cluster.
+
+If a data stream is restored, its backing indices are also restored. However, restored backing indices are not automatically
+added to any existing data streams. For example, if only the
+`.ds-logs-000003` backing index is restored from a snapshot, it is not automatically added to the existing `logs` data stream.
+
+include::{es-ref-dir}/snapshot-restore/restore-snapshot.asciidoc[tag=index-settings-data-stream-warning]
+
+[[restore-snapshot-api-path-params]]
+==== {api-path-parms-title}
+
+`<repository>`::
+(Required, string)
+Name of the repository to restore a snapshot from.
+
+`<snapshot>`::
+(Required, string)
+Name of the snapshot to restore. This name must be unique in the snapshot repository.
+
+[role="child-attributes"]
+[[restore-snapshot-api-request-body]]
+==== {api-request-body-title}
+
+`ignore_index_settings`::
+(Optional, string)
+A comma-separated list of index settings and index settings groups that should not be restored from a snapshot.
+
+`include_aliases`::
+(Optional, boolean)
+If `true`, aliases from the original snapshot will be restored.
+Defaults to `true`.
++
+If `false`, prevents aliases from being restored together with associated
+indices.
+
+[[restore-snapshot-api-include-global-state]]
+`include_global_state`::
++
+--
+(Optional, boolean)
+If `false`, the cluster state is not restored. Defaults to `false`.
+
+If `true`, the current cluster state is included in the restore operation.
+
+The cluster state includes:
+
+* Persistent cluster settings
+* Index templates
+* Legacy index templates
+* Ingest pipelines
+* {ilm-init} lifecycle policies
+--
++
+IMPORTANT: By default, the entire restore operation will fail if one or more indices included in the snapshot do not have all primary shards available. You can change this behavior by setting <<restore-snapshot-api-partial,`partial`>> to `true`.
```

Comment:
I'm not sure how this relates to the `include_global_state` param.

It's also not clear that the primary shards have to be available _when the snapshot_ is taken. As-is, it could be read as meaning an existing index needs to have its primary shards available.